llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -379,7 +379,7 @@ static bool mergeBlocksIntoPredecessors(VPlan &Plan) {
     // Don't fold the exit block of the Plan into its single predecessor for
     // now.
     // TODO: Remove restriction once more of the skeleton is modeled in VPlan.
-    if (VPBB->getNumSuccessors() == 0 && !VPBB->getParent())
+    if (!VPBB->getParent())
```

Comment:
The condition for zero successors is dropped because it now coincides with (scalar header and exit) having no parental region? Above comment about refraining from folding exit block "for now" still intact?

Reply:
Yep, updated to skip this for any block in the skeleton.